app/upgrades/v18/upgrades.go

```diff
@@ -0,0 +1,216 @@
+			"1706122819": NewIntFromString("3634"),
+			"1706554818": NewIntFromString("19330705153"),
+			"1706986821": NewIntFromString("21799461066"),
+			"1707418817": NewIntFromString("140098739"),
+			"1707850817": NewIntFromString("7879600093"),
+		},
+		"stargaze-1": {
+			"1705690816": NewIntFromString("327679537357"),
+			"1705950009": NewIntFromString("437989561516"),
+			"1706209215": NewIntFromString("437233672994"),
+			"1706468412": NewIntFromString("508229473185"),
+			"1706727610": NewIntFromString("1033532361858"),
+		},
+		"umee-1": {
+			"1705690815": NewIntFromString("510084698639"),
+		},
+	}
+)
+
+// CreateUpgradeHandler creates an SDK upgrade handler for v17
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configurator module.Configurator,
+	stakeibcKeeper stakeibckeeper.Keeper,
+	recordsKeeper recordskeeper.Keeper,
+) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		ctx.Logger().Info("Starting upgrade v18...")
+
+		ctx.Logger().Info("Updating redemption rate bounds...")
+		UpdateRedemptionRateBounds(ctx, stakeibcKeeper)
+
+		ctx.Logger().Info("Updating unbonding records...")
+		err := UpdateUnbondingRecords(ctx, stakeibcKeeper, recordsKeeper)
+		if err != nil {
+			return vm, err
+		}
+
+		return mm.RunMigrations(ctx, configurator, vm)
+	}
+}
+
+// Updates the outer redemption rate bounds
+func UpdateRedemptionRateBounds(ctx sdk.Context, k stakeibckeeper.Keeper) {
+	for _, hostZone := range k.GetAllHostZone(ctx) {
+		// Give osmosis a bit more slack since OSMO stakers collect real yield
+		outerAdjustment := RedemptionRateOuterMaxAdjustment
+		if hostZone.ChainId == OsmosisChainId {
+			outerAdjustment = outerAdjustment.Add(OsmosisRedemptionRateBuffer)
+		}
+
+		outerMinDelta := hostZone.RedemptionRate.Mul(RedemptionRateOuterMinAdjustment)
+		outerMaxDelta := hostZone.RedemptionRate.Mul(outerAdjustment)
+
+		outerMin := hostZone.RedemptionRate.Sub(outerMinDelta)
+		outerMax := hostZone.RedemptionRate.Add(outerMaxDelta)
+
+		hostZone.MinRedemptionRate = outerMin
+		hostZone.MaxRedemptionRate = outerMax
+
+		k.SetHostZone(ctx, hostZone)
+	}
+}
+
+// Modify HostZoneUnbonding and UserRedemptionRecords NativeTokenAmount to reflect new data structs
+func UpdateUnbondingRecords(ctx sdk.Context, k stakeibckeeper.Keeper, rk recordskeeper.Keeper) error {
+	for _, epochUnbondingRecord := range rk.GetAllEpochUnbondingRecord(ctx) {
+		// loop over host zone unbonding records
+		for _, hostZoneUnbonding := range epochUnbondingRecord.GetHostZoneUnbondings() {
+			recordStatus := hostZoneUnbonding.Status
+			if recordStatus != recordtypes.HostZoneUnbonding_EXIT_TRANSFER_QUEUE {
+				continue // can ignore this record, as it's not unbonding
+			}
+			amountUnbonded := sdkmath.ZeroInt()
+			// check if the unbond amount is stored in cache above
+			allAmountsUnbondedForHost, found := TotalAmountUnbonded[hostZoneUnbonding.HostZoneId]
+			if !found {
```

Comment:
is this supposed to be `found` not `!found`?

Reply:
🤦  Yes definitely, that was a mistake on my part